wicker/core/column_files.py

```diff
@@ -182,16 +183,49 @@ def _write_column(self, column_name: str) -> None:
         self.write_buffers[column_name] = self._get_new_buffer()
 
 
-class ColumnBytesFileCache:
+class ColumnBytesFileReader(abc.ABC):
+    """A reader class for column bytes files to inherit upon for caches and cacheless"""
+
+    def __init__(
+        self,
+        column_bytes_root_path: str,
+    ) -> None:
+        super().__init__()
+        self._column_bytes_root_path = column_bytes_root_path
+
+    def read(self, column_bytes_file_info: ColumnBytesFileLocationV1) -> bytes:
+        """Read data from column bytes file.
+
+        :param column_bytes_file_info: Location object to column bytes file on data store
+        :type column_bytes_file_info: ColumnBytesFileLocationV1
+        :return: bytes read from the column bytes file
+        """
+        column_concatenated_bytes_file_path = os.path.join(
+            self._column_bytes_root_path, str(column_bytes_file_info.file_id)
+        )
+
+        with open(column_concatenated_bytes_file_path, "rb") as f:
+            f.seek(column_bytes_file_info.byte_offset)
+            return f.read(column_bytes_file_info.data_size)
+
+    def resolve_pointers(
+        self,
+        example: validation.AvroRecord,
+        schema: schema.DatasetSchema,
+    ):
+        visitor = ResolvePointersVisitor(example, schema, lambda cbf_info: self.read(cbf_info))
```

Comment:
@isaak-willett I have a preference to actually pass the `ColumnBytesFileReader` (i.e. `self`) to the `ResolvePointersVisitor` as its last argument (instead of passing a `lambda`). Then the visitor can just call the `read` function on the reader.

Reply:
Done